### ISSUE TITLE
MacOS: Window resizing causes the whole UI to stretch 

### ISSUE THREAD

**Issue by eliheuer** [MFEK/glif#322]
See the screenshot below, this seems distracting and hopefully unnecessary. MacOS 11
<img width="1920" alt="stretch" src="https://user-images.githubusercontent.com/5162664/185779594-6de8d943-96f1-438c-b36e-e94c4e2221dc.png">

**Comment by ctrlcctrlv** [comment]
Uh oh. How do we think this happened @Subject38? Might you have time to test with me via TeamViewer again some time soon?

**Comment by ctrlcctrlv** [comment]
Did I split the scale into (x,y) components and forget a `min()`? That's my first thought but I think we would have caught it so it might be more Apple fuckery, they cannot seem to keep their API's straight.

**Comment by ctrlcctrlv** [comment]
Also if Seth has no time @eliheuer can test with me just as well. I could not find an affordable macOS virtualization service _with GPU support_ so have not been able to determine another way to test, and the Foundation is (I am) naturally quite against the idea of purchasing an Apple computer to do local testing.

**Comment by eliheuer** [comment]
@ctrlcctrlv It is a bit hard to see what is going on in just a screenshot, so I made a YouTube video showing the bug here: https://www.youtube.com/watch?v=VzSWypCgkdQ

This is on a 2020 Apple M1 Mac Mini, macOS 11.6.5.

**Comment by eliheuer** [comment]
On Arch Linux resizing the window makes MFEKglif crash before I ever see any stretching, but this seems like another bug in itself. 
```
This is MFEKglif v1.1.1b2-25-gc874d5a (“Metafontverse”), compiled @ 2022年08月21日(日)　16時43分57秒(午後)　協定世界時-0700.
[2022-08-21T23:51:44Z ERROR mfek_ipc::module] Module MFEKmetadata is not available. MFEK is modular software; it will still run but some features will not be available. For the best experience, please install all available MFEK modules into your PATH.
[2022-08-21T23:51:44Z INFO  MFEKglif::settings] Configuration directory is "/home/eli/.config/MFEK/glif"
[2022-08-21T23:51:44Z WARN  MFEKglif::command] Keybinds XML invalid (Invalid command string!), using default
[2022-08-21T23:51:57Z WARN  MFEKglif::user_interface] Failed to draw frame. This can happen when resizing due to VkError(ERROR_DEVICE_LOST); if happens otherwise, file an issue.

panicked at 'called `Option::unwrap()` on a `None` value', /home/eli/.cargo/registry/src/github.com-1ecc6299db9ec823/rafx-api-0.0.14/src/extra/swapchain_helper.rs:347:14

[2022-08-21T23:51:57Z INFO  MFEKglif::util] Wrote backtrace to /tmp/error_log_1661125917.txt
[1]    9222 abort (core dumped)  cargo run --release -- examples/Q_.glif
```

**Comment by eliheuer** [comment]
I just tested on Arch Linux and was able to resize the window without any stretching of the UI, video example here: https://www.youtube.com/watch?v=ZmyjAzNq7hs

So this is a MacOS bug, happy to help look into this, I'll see if I get the same thing on a older Intel non-M1 mac next.

**Comment by ctrlcctrlv** [comment]
I have determined the fastest way to fixing Mac issues is for me to just use someone else's Mac PC and try a bunch of things until I fix it. 

Thanks for showing me the issue, it's just as weird as I thought. For sure @Subject38 and I would have noticed this. So, I can't really fix it until someone has time to dedicate to a remote session.

Because I also tried with @Subject38 to just send patches I *thought* might work in the past and it was a ton of useless work for both of us, in terms of patch management, copying, pasting, it not working, me getting back a video, etc.

**Comment by ctrlcctrlv** [comment]
Humorous new tag:

![image](https://user-images.githubusercontent.com/838783/185823973-943b042f-54c5-4687-8e2f-61f3b802835d.png)

**Comment by eliheuer** [comment]
I updated to macOS 12.5.1 tonight and I still get the same stretching when resizing the window. So macOS 11 was not the issue.

**Comment by Subject38** [comment]
Just tested on my local machine and I can confirm this issue exists. I feel like I might have noticed it before but didn't realise it was a big issue since after resizing all the UI elements go back to normal. I'm honestly not entirely sure what might be causing this behavior in the first place...

**Comment by ctrlcctrlv** [comment]
> after resizing all the UI elements go back to normal

Oh so this isn't even that big a deal but @Subject38 there's a bigger problem

I'm glad I tried to do a full build again on @t0lva's PC because I literally was on the phone w/her for an hour trying to build and it kept erroring out and it turns out we're blocked on https://github.com/aclysma/skulpin/issues/108#issuecomment-1221852846 for new VulkanSDK versions

What VulkanSDK do you have?

**Comment by ctrlcctrlv** [comment]
(That's @t01va not @t0lva.)

**Comment by ctrlcctrlv** [comment]
(Housekeeping: I'm downgrading this bug to minor and making a new one for the VulkanSDK update issue.)

**Comment by Subject38** [comment]
```
sub38@fear ~ % vulkaninfo | grep Version
WARNING: [MoltenVK] Code 0 : Unrecognized CreateInstance->pCreateInfo->pApplicationInfo->apiVersion number (0x004020c6). Assuming MoltenVK 1.1 version.
Vulkan Instance Version: 1.2.198
```
Looks like version 1.2.198.
I never updated after initially installing so that's likely why...

**Comment by ctrlcctrlv** [comment]
Thank you that jives perfectly w/@eliheuer's experience. I opened #323 for the SDK issue.

For the resizing...it's basically #295. Sorry I just don't think it's a big deal but of course it's still a bug so I'm leaving it open.

**Comment by eliheuer** [comment]
I just tested this on an Intel mac on macOS 11 and still get the same stretching.

I think from a UI/UX perspective it is a major issue, but I'm a UI snob I guess.

**Comment by ctrlcctrlv** [comment]
@eliheuer I agree it looks really wonky and ugly (same with the blackout effect that happens on Linux). I'll be really honest, I just don't know _how_ to fix it. Like there are callbacks and Vulkan something or other going on here, it's all wrapped up with Skulpin's Vulkan renderer in general and I am just as likely to make it worse as make it better.

@MatthewBlanchard is usually our 3D expert but he's been a bit MIA of late so if he doesn't fix it I don't know when/if it will be, and personally I don't see it as a major problem compared to all the other ones, so that's why I've tagged it `low-priority`. That doesn't mean it needs to be low priority for you or anyone else including Matt, the tags just reflect my thinking as this project's maintainer. Anyone else is free to pitch in. :-)

**Comment by ctrlcctrlv** [comment]
This issue has been merged with, and supersedes, #295.

**Comment by jakeisnt** [comment]
By the blackout effect, are y'all referring to the flickering/resetting effect that I receive? [Screen recording](https://youtu.be/UCcV2kBe0IU) to demonstrate the behavior on my system (Linux/NixOS/Wayland, using native Wayland, not XWayland). I don't see any stretching or shrinking unfortunately.

**Comment by ctrlcctrlv** [comment]
Yes, the issues are one and the same and have the same root upstream cause.

**Comment by ctrlcctrlv** [comment]
It's worth noting what I wrote at https://github.com/imgui-rs/imgui-rs/pull/582#issuecomment-1363584268. To fix #323 we will be switching in the new year to @lucasmerlin's [egui_skia](https://github.com/lucasmerlin/egui_skia/), OpenGL by default, with Vulkan optionally, across all GUI MFEK modules like this one.

That might fix this; in my tests so far it does. It also might not. But @jackeisnt the core GUI rendering has such large planned changes I don't think it's worth fixing this issue unless it's through the planned upgrade path. :-)

**Comment by eliheuer** [comment]
I'm very happy to hear about egui_skia and the big changes to the GUI rendering. Thanks!
